src/components/ToggleButton/ToggleButton.tsx

```diff
@@ -1,31 +1,38 @@
-import { useState } from "react";
+import { useToggle } from "hooks/useToggle";
+import { ButtonHTMLAttributes, DetailedHTMLProps, ReactNode, useEffect, MouseEvent } from "react";
 import style from "./ToggleButton.module.scss"
 
 type Props = {
     label: string;
-    id?: string;
-    icon: React.ReactNode;
-    onToggle: (target: string, state: boolean) => void;
-}
+    icon: ReactNode;
+    onToggle?: (state: boolean) => void;
+} & DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>
 
-export function ToggleButton(props: Props) {
-    const [isToggeled, setIsToggeled] = useState(false)
+export function ToggleButton({ label, icon, onToggle, onClick, ...buttonProps }: Props) {
+    const [isOn, flip] = useToggle()
 
-    const target = props.id ?? props.label
-    function onClick() {
-        setIsToggeled((prev) => {
-            const next = !prev
-            props.onToggle(target, next)
-            return next
-        })
+    if (onToggle) {
```

Comment:
Conditional hooks are not allowed, do

`useEffect(()=>{
onToggle?.() // This syntax is like conditional calling the function if is not undefined
}, [isOn])`